test/bugs/issue_549.test.ts

```diff
@@ -52,23 +52,23 @@ describe("Issue 549", () => {
     try {
       const result = willThrow();
       throw new Error(
-        `This line should never be executed. Expected 'willThrow' to throw! ${JSON.stringify(result)}`
+        `This line should never be executed. Expected \`willThrow\` to throw! ${JSON.stringify(result)}`
       );
     } catch (e) {
-
-      const expectedErrorA = ERROR_MSGS.CIRCULAR_DEPENDENCY_IN_FACTORY("toDynamicValue", TYPE.ADynamicValue.toString());
-      const expectedErrorB = ERROR_MSGS.CIRCULAR_DEPENDENCY_IN_FACTORY("toDynamicValue", TYPE.BDynamicValue.toString());
-      const matchesErrorA = e.message.indexOf(expectedErrorA) !== -1;
-      const matchesErrorB = e.message.indexOf(expectedErrorB) !== -1;
+      const localError = e as Error;
+      const expectedErrorA = ERROR_MSGS.CIRCULAR_DEPENDENCY_IN_FACTORY('toDynamicValue', TYPE.ADynamicValue.toString());
+      const expectedErrorB = ERROR_MSGS.CIRCULAR_DEPENDENCY_IN_FACTORY('toDynamicValue', TYPE.BDynamicValue.toString());
+      const matchesErrorA = localError.message.indexOf(expectedErrorA) !== -1;
+      const matchesErrorB = localError.message.indexOf(expectedErrorB) !== -1;
```

Comment:
Not sure if this is better or not
```suggestion
      const matchesErrorA = (e as Error).message.indexOf(expectedErrorA) !== -1;
      const matchesErrorB = (e as Error).message.indexOf(expectedErrorB) !== -1;
```

Reply:
I think the existing version has less redundancy, I would recommend against this suggestion